binding.cc

```diff
@@ -1015,6 +1025,86 @@ namespace zmq {
 
 #endif
 
+  NAN_METHOD(Socket::Readv) {
+    Socket* socket = GetSocket(info);
+    if (socket->state_ != STATE_READY)
+      return info.GetReturnValue().Set(Nan::Undefined());
+
+    int events;
+    size_t events_size = sizeof(events);
+
+    int rc = 0;
+    int flags = 0;
+    int64_t more = 1;
+    size_t more_size = sizeof(more);
+    size_t index = 0;
+
+    while (true) {
```

Comment:
I'm not too fond of this way of structuring the loop. A do-while loop with a flag is clearer than an infinite loop with a `break` and a conditional `continue` to avoid the break.